### ISSUE TITLE
ArgumentNullException in LocalTransport.InitStreams when using MIMode = gdb

### ISSUE THREAD

**Issue by puremourning** [microsoft/MIEngine#911]
When attempting to attach to a process on macOS using "MIMode": "gdb", the following error is thrown:

<img width="1363" alt="Screenshot 2019-08-10 at 14 46 30" src="https://user-images.githubusercontent.com/10584846/62822599-beecb200-bb7d-11e9-93d4-7bb7c4876bcf.png">

Steps to reproduce in VScode:

* Create a project and use the "(lldb) Attach" template
* Change MIMode to gdb
* Attach to a program.

```
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "configurations": [
    { 
      "name": "(lldb) Attach",
      "type": "cppdbg",
      "request": "attach",
      "program": "${workspaceRoot}/build-debug/bin/clangd",
      "processId": "${command:pickProcess}",
      "MIMode": "gdb",
      "logging": { "engineLogging": true }
    }

  ]
}
```

I have reproduced this in both VSCode and Vimspector (another Debug Adapter Protocol client).

In the vimspector case, we're certainly sending 'cwd' key:

```
2019-08-10 14:53:19,915 - DEBUG - Sending Message: {"command": "attach", "arguments": {"type": "cppdbg", "request": "attach", "program": "/Users/ben/Development/llvm-project/clang-tools-extra/clangd/../../build-debug/bin/clangd", "cwd": "/Users/ben/Development/llvm-project/clang-tools-extra/clangd", "args": [], "env": {"LD_LIBRARY_PATH": "/Users/ben/Development/llvm-project/clang-tools-extra/clangd/../../build-debug/lib"}, "miMode": "gdb", "stopAtEntry": false, "logging": {"engineLogging": true}, "processId": "62648", "name": "test"}, "seq": 1, "type": "request"}
```

Full error including the DAP log in this gist: https://gist.github.com/puremourning/35e2340ed364576b7721e3885a923445

Seems to have been reported before: https://github.com/microsoft/vscode-cpptools/issues/548#issuecomment-324781355

**Comment by puremourning** [comment]
FWIW this is the version of vscode-cpptools that's installed in viscose:

```
Name: C/C++
Id: ms-vscode.cpptools
Description: C/C++ IntelliSense, debugging, and code browsing.
Version: 0.25.0-insiders
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools
```

**Comment by pieandcakes** [comment]
@puremourning The problem is that if you specify `"MIMode": "gdb"` you need to specify `"miDebuggerPath"` to your location of `gdb`. I'll need to debug it and figure out what changed to cause it to throw the exception instead.

**Comment by pieandcakes** [comment]
The actual issue of this was because `cwd` is also empty and was fixed in #893. It will be in the next release.

**Comment by sean-mcmanus** [comment]
Should be fixed with 0.25.0. @pieandcakes Can you close this?
